proc/variables.go

```diff
@@ -520,16 +512,16 @@ func (thread *Thread) extractValueInternal(instructions []byte, addr int64, typ
 	case *dwarf.StructType:
 		switch {
 		case t.StructName == "string":
-			return thread.readString(ptraddress)
+			return scope.Thread.readString(ptraddress)
 		case strings.HasPrefix(t.StructName, "[]"):
-			return thread.readSlice(ptraddress, t, recurseLevel)
+			return scope.readSlice(ptraddress, t, recurseLevel)
```

Comment:
readSlice calls extractValue therefore it needs scope, everything I could leave on thread I left there, readIntRaw doesn't need a scope, just an address.

When the code that calculates addresses is separated from the code that reads memory it will be possible to clean the interface further (the methods reading memory can all be on Thread, the ones calculating addresses need EvalScope).